tests/zmq/test_gnmi_zmq.py

```diff
@@ -0,0 +1,130 @@
+
+    # swss and gnmi container may take some time to stop after reload config command
+    time.sleep(5)
+
+    pytest_assert(wait_until(30, 2, 0, _check_process_ready, "orchagent"),
+                  "The orchagent not start after change subtype")
+
+    pytest_assert(wait_until(30, 2, 0, _check_process_ready, "telemetry"),
+                  "The telemetry not start after change subtype")
+
+
+@pytest.fixture
+def enable_zmq(duthost):
+    command = 'sonic-db-cli CONFIG_DB hget "DEVICE_METADATA|localhost" subtype'
+    subtype = duthost.shell(command, module_ignore_errors=True)["stdout"]
+    logger.debug("subtype: {}".format(subtype))
+
+    # the device already enable SmartSwitch
+    if subtype == "SmartSwitch":
+        yield
+        return
+
+    # enable ZMQ
+    command = 'sonic-db-cli CONFIG_DB hset "DEVICE_METADATA|localhost" subtype SmartSwitch'
+    result = duthost.shell(command, module_ignore_errors=True)
+    logger.debug("set subtype subtype: {}".format(result))
+    save_reload_config(duthost)
+
+    yield
+
+    # revert change
+    command = 'sonic-db-cli CONFIG_DB hdel "DEVICE_METADATA|localhost" subtype'
+    result = duthost.shell(command, module_ignore_errors=True)
+    logger.debug("revert subtype subtype: {}".format(result))
+    save_reload_config(duthost)
+
+
+def gnmi_set(duthost, ptfhost, delete_list, update_list, replace_list):
+    ip = duthost.mgmt_ip
+    port = 8080
+    cmd = 'python2 /root/gnxi/gnmi_cli_py/py_gnmicli.py '
+    cmd += '--timeout 30 --notls '
+    cmd += '--notls '
+    cmd += '-t %s -p %u ' % (ip, port)
+    cmd += '-xo sonic-db '
+    cmd += '-m set-update '
+    xpath = ''
+    xvalue = ''
+    for path in delete_list:
+        path = path.replace('sonic-db:', '')
+        xpath += ' ' + path
+        xvalue += ' ""'
+    for update in update_list:
+        update = update.replace('sonic-db:', '')
+        result = update.rsplit(':', 1)
+        xpath += ' ' + result[0]
+        xvalue += ' ' + result[1]
+    for replace in replace_list:
+        replace = replace.replace('sonic-db:', '')
+        result = replace.rsplit(':', 1)
+        xpath += ' ' + result[0]
+        if '#' in result[1]:
+            xvalue += ' ""'
+        else:
+            xvalue += ' ' + result[1]
+    cmd += '--xpath ' + xpath
+    cmd += ' '
+    cmd += '--value ' + xvalue
+    output = ptfhost.shell(cmd, module_ignore_errors=True)
+    error = "GRPC error\n"
+    if error in output['stdout']:
+        result = output['stdout'].split(error, 1)
+        raise Exception("GRPC error:" + result[1])
+    if output['stderr']:
+        raise Exception("error:" + output['stderr'])
+    else:
+        return
+
+
+def test_gnmi_zmq(duthosts,
+                  rand_one_dut_hostname,
+                  ptfhost,
+                  enable_zmq):
+    duthost = duthosts[rand_one_dut_hostname]
+
+    command = 'ps -auxww | grep "/usr/sbin/telemetry -logtostderr --noTLS --port 8080"'
```

Comment:
If telemetry isn't running as expected, this test will fail

Reply:
enable_zmq fixture will make the telemetry run with ZMQ enabled.
The debug info here will collect the telemetry parameters for investigation.